File: demo-grpc-gradle-new/demo-grpc-gradle-new-consumer/src/main/java/com/example/demo/grpc/command/MessageServiceCommander.java
package com.example.demo.grpc.command;

import com.example.demo.grpc.client.MessageServiceClient;
import com.example.demo.grpc.proto.Message;
import com.google.common.util.concurrent.ListenableFuture;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import java.nio.charset.StandardCharsets;

@Component
public class MessageServiceCommander implements CommandLineRunner {
    @Autowired
    private MessageServiceClient messageServiceClient;
    @Override
    public void run(String... args) throws Exception {
        ListenableFuture<Message.MessageResponse> future = messageServiceClient.sendMessage("aaa", "aaa", "aaa");
        Message.MessageResponse messageResponse = future.get();
        String s = messageResponse.toString();
        String string = messageResponse.getMessageBytes().toString(StandardCharsets.UTF_8);
        System.out.println(s);
        System.out.println("string = " + string);
    }
}

File: demo-kotlin-grpc-maven/demo-kotlin-grpc-maven-consumer/src/main/kotlin/com/example/demo/grpc/command/MessageServiceCommander.kt
package com.example.demo.grpc.command

import com.example.demo.grpc.client.MessageServiceClient
import com.example.demo.grpc.proto.Message.MessageRequest
import kotlinx.coroutines.flow.onEach
import kotlinx.coroutines.flow.toList
import kotlinx.coroutines.runBlocking
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class MessageServiceCommander:CommandLineRunner {
    @Autowired
    private lateinit var messageServiceClient: MessageServiceClient
    override fun run(vararg args: String?) {
        val runBlocking: String = runBlocking {
            val sendMessage = messageServiceClient.sendMessage("aaa", "aaa", "aaa")
            val string = sendMessage.messageBytes.toString("UTF-8")
            sendMessage.success.toString() + string
        }
        println(runBlocking)

        val messageRequest = ArrayList<MessageRequest>()
        for (i in 1..1000){
            messageRequest.add(MessageRequest.newBuilder().setContent("").setReceiver(UUID.randomUUID().toString()).build())
        }
        val runBlocking1 = runBlocking {
            val sendMessageFlow = messageServiceClient.sendMessageFlow(messageRequest)
            sendMessageFlow.toList()
        }
        runBlocking1.onEach { messageResponse -> println(messageResponse.success.toString()+messageResponse.messageBytes.toString("UTF-8")) }
    }
}